cpp/thirdparty/versions.txt

```diff
@@ -64,10 +64,10 @@ ARROW_MIMALLOC_BUILD_VERSION=v2.0.6
 ARROW_MIMALLOC_BUILD_SHA256_CHECKSUM=9f05c94cc2b017ed13698834ac2a3567b6339a8bde27640df5a1581d49d05ce5
 ARROW_NLOHMANN_JSON_BUILD_VERSION=v3.10.5
 ARROW_NLOHMANN_JSON_BUILD_SHA256_CHECKSUM=5daca6ca216495edf89d167f808d1d03c4a4d929cef7da5e10f135ae1540c7e4
-ARROW_OPENTELEMETRY_BUILD_VERSION=v1.3.0
-ARROW_OPENTELEMETRY_BUILD_SHA256_CHECKSUM=6a4c43b9c9f753841ebc0fe2717325271f02e2a1d5ddd0b52735c35243629ab3
-ARROW_OPENTELEMETRY_PROTO_BUILD_VERSION=v0.11.0
-ARROW_OPENTELEMETRY_PROTO_BUILD_SHA256_CHECKSUM=985367f8905e91018e636cbf0d83ab3f834b665c4f5899a27d10cae9657710e2
+ARROW_OPENTELEMETRY_BUILD_VERSION=v1.4.1
+ARROW_OPENTELEMETRY_BUILD_SHA256_CHECKSUM=301b1ab74a664723560f46c29f228360aff1e2d63e930b963755ea077ae67524
+ARROW_OPENTELEMETRY_PROTO_BUILD_VERSION=v0.18.0
```

Comment:
Technically upstream is on 0.17.0: https://github.com/open-telemetry/opentelemetry-cpp/tree/v1.4.1/third_party

Reply:
Pushed https://github.com/apache/arrow/pull/13580/commits/8628ae007e1661048cbd25302286ee0010f03ff2 to match version used on opentelemetry-cpp upstream